examples/keras_imagenet_resnet50.py

```diff
@@ -53,55 +54,63 @@
-    zoom_range=(0.875, 0.875), preprocessing_function=keras.applications.resnet50.preprocess_input)
+    zoom_range=(0.875, 0.875),
+    preprocessing_function=keras.applications.resnet50.preprocess_input)
 test_iter = test_gen.flow_from_directory(test_dir, batch_size=batch_size,
                                          target_size=(224, 224))
 
-# Set up standard ResNet-50 model.
-model = keras.applications.resnet50.ResNet50(weights=None)
-
-# ResNet-50 model that is included with Keras is optimized for inference.
-# Add L2 weight decay & adjust BN settings.
-model_config = model.get_config()
-for layer, layer_config in zip(model.layers, model_config['layers']):
-    if hasattr(layer, 'kernel_regularizer'):
-        regularizer = keras.regularizers.l2(weight_decay)
-        layer_config['config']['kernel_regularizer'] = \
-            {'class_name': regularizer.__class__.__name__,
-             'config': regularizer.get_config()}
-    if type(layer) == keras.layers.BatchNormalization:
-        layer_config['config']['momentum'] = 0.9
-        layer_config['config']['epsilon'] = 1e-5
-model = keras.models.Model.from_config(model_config)
-
-# Horovod: adjust learning rate based on number of GPUs.
-opt = keras.optimizers.SGD(lr=learning_rate * hvd.size(), momentum=0.9)
-
-# Horovod: add Horovod Distributed Optimizer.
-opt = hvd.DistributedOptimizer(opt)
-
-model.compile(loss=keras.losses.categorical_crossentropy,
-              optimizer=opt,
-              metrics=['accuracy', 'top_k_categorical_accuracy'])
+if resume_from_epoch == 0:
+    # Set up standard ResNet-50 model.
+    model = keras.applications.resnet50.ResNet50(weights=None)
+
+    # ResNet-50 model that is included with Keras is optimized for inference.
+    # Add L2 weight decay & adjust BN settings.
+    model_config = model.get_config()
+    for layer, layer_config in zip(model.layers, model_config['layers']):
+        if hasattr(layer, 'kernel_regularizer'):
+            regularizer = keras.regularizers.l2(weight_decay)
+            layer_config['config']['kernel_regularizer'] = \
+                {'class_name': regularizer.__class__.__name__,
+                 'config': regularizer.get_config()}
+        if type(layer) == keras.layers.BatchNormalization:
+            layer_config['config']['momentum'] = 0.9
+            layer_config['config']['epsilon'] = 1e-5
+    model = keras.models.Model.from_config(model_config)
+
+    # Horovod: adjust learning rate based on number of GPUs.
+    opt = keras.optimizers.SGD(lr=learning_rate * hvd.size(), momentum=0.9)
+
+    # Horovod: add Horovod Distributed Optimizer.
+    opt = hvd.DistributedOptimizer(opt)
+
+    model.compile(loss=keras.losses.categorical_crossentropy,
+                  optimizer=opt,
+                  metrics=['accuracy', 'top_k_categorical_accuracy'])
+else:
+    # Restore from a previous checkpoint, if initial epoch is specified.
+    model = hvd.load_model(checkpoint_format.format(epoch=resume_from_epoch))
```

Comment:
Only rank-0 should be required to read the file, as per:
```
Horovod: restore on the first worker which will broadcast weights to other workers.
```

Reply:
Does that also broadcast the optimizer weights and state as well?

Reply:
If they are TF global variables, it should - but would be great to check.